### ISSUE TITLE
MONGO_URL with multiple hosts throws useless error if any of them are invalid.

### ISSUE THREAD

**Issue by jasonk** [meteor/meteor#8147]
I haven't completely tracked down what is causing it, but since I just spent the better part of two days figuring out what the problem actually was I figured I should get it reported..

When starting Meteor with a MONGO_URL that includes more than one mongodb server, if one of them is invalid then the error message that results is entirely unhelpful, consisting of just (yes, this is indeed the entire error):

```
/Users/jasonkohles/.meteor/packages/meteor-tool/.1.4.2_3.bdala++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:313
            throw(ex);
            ^
[object Object]
```

I'm running Meteor 1.4.2.3, and have confirmed that you can very easily reproduce this with pretty much any meteor app:

```
git clone https://github.com/meteor/todos
cd todos
meteor npm install
export MONGO_URL=mongodb://foo,bar/blah
meteor
# (here Meteor crashes with [object Object] error message)
```

**Comment by emrekeskinmac** [comment]
I am doing this and reaching export "MONGO_URL" = "mongodb: //255.255.255.255: 27017 / meteor"

**Comment by jasonk** [comment]
I suspect it's related to how the URLs are parsed.  The error message is fine if your MONGO_URL only has one host in it, but breaks down if it has more than one and some of them are invalid.  I ran into this because of a deployment script that assumed that the environment would have a 3-node mongo cluster, but in dev we only had a single server, so the other two didn't resolve.

Also, in the example I included, the "foo,bar/blah" isn't a placeholder to fill in, if you try to start Meteor with that literal URL you'll see the error.  I've reproduced it that way both on OS X and RHEL 7..

**Comment by abernix** [comment]
This is essentially the same issue as #7954 and is an uncaught exception by the [MongoDB Node Driver (`mongodb` NPM)](https://github.com/mongodb/node-mongodb-native).  Essentially what you're seeing is an unserializable `EventEmitter` object.

This was actually fixed in an upstream Node issue ([Jira NODE-838](https://jira.mongodb.org/browse/NODE-838), where you can see a non-Meteor way to reproduce it) and released in `mongodb@2.2.12`.  Meteor just hasn't updated to use that version yet.

I've opened PR #8150.

**Comment by hwillson** [comment]
If I'm not mistaken, this issue has been fixed by PR #8179 (which was merged). Closing - please post back if this is still happening. Thanks for posting this originally!
